initial attempt at planet data converter

cholmes opened the pull request:
This one needs a bit of help, but I think is pretty close.

I got it mostly working, though couldn't figure out why it was using 0.4.0 of the CLI, until I pushed the branch and realized I'd accidentally branched off of the Austria and Brazil converter branch that I was investigating. But I had managed to get it working locally.

Then I switched to 0.5.0, pleased to remove my dummy 'bbox' that I stuck in, but it's now stuck. I get:

```
% fiboa convert planet_afb -o ~/repos/data-fiboa/output/planet.parquet
fiboa CLI 0.5.0 - Convert 'planet_afb'

Getting file(s) if not cached yet
'dict' object has no attribute 'delete'
```

I couldn't find where the 'delete' was coming from, or where it should be. Any ideas? 

This PR also raises some issues, I mostly highlighted as comments in the code, the main one being that this is not about converted a fixed, online dataset, but is meant to be a tool to converter any Planet field boundary output. So it'd ideally have like a `-i` flag to point at a file instead of a default online location, or even require an input argument for this conversion. So curious how to approach this. It will be similar for DigiFarm and OneSoil, two other converters I'd like to make, as those are the three main field boundary providers that I know of.

m-mohr commented:
The delete issue has been fixed, I think. It only appears with Python 3.12 or later, which I don't have yet.

With the changes relating to extracting ZIP files, handling multiple input files, and caching, it's not overly easy anymore to pass in the source files easily. The -c option which we've used before now accepts a folder with files included and then the converter needs to specify the filenames in the folders. Also, a -i option could only handle the simple usecase of a non-extractable files.
Have a look at be_wa, which has also external data that you need to pass in from an external source. There's a DATA_ACCESS flag, which you can fill with instructions and then you can use -c and SOURCES in combination. Would that work for you?

cholmes commented:
Ok, I checked the improvements, they all look great. I updated the docs on this a bit, and also added determination method. So I think it's ready to merge.

thanks!
